token store expiration failed to revoke non SSCT (missing cubbyhole ID)

<!-- Please reserve GitHub issues for bug reports and feature requests.

For questions, the best place to get answers is on our [discussion forum](https://discuss.hashicorp.com/c/vault), as they will get more visibility from experienced users than the issue tracker.

Please note: We take Vault's security and our users' trust very seriously. If you believe you have found a security issue in Vault, please responsibly disclose by contacting us at security@hashicorp.com. Our PGP key is available at [our security page](https://www.hashicorp.com/security/).

-->

**Describe the bug**
Vault Server unable to revoke old tokens (27 bytes) non SSCT.
2022-12-05T20:27:51.825Z [ERROR] expiration: failed to revoke lease: lease_id=auth/cert/login/h6d0772972fd1601be0b77048fb70dd7bb217982fefd4f0eb41c2fba8b49bc188 error="failed to revoke token: failed to revoke entry: missing cubbyhole ID while destroying"  


**To Reproduce**
1. Run Vault server and set VAULT_DISABLE_SERVER_SIDE_CONSISTENT_TOKENS to true 
2. vault login <root token>
3. vault auth enable cert
4. vault write auth/cert/certs/web \
    display_name=web \
    policies=web,prod \
    certificate=@web-cert.pem \
    ttl=60
5. Run multiple login curl -k https://localhost:8200/v1/auth/cert/login -X POST --cert ./web-cert.pem --key ./web-key.pem | jq
6. wait a 1-2 min 
7. check the logs 
8. [ERROR] expiration: failed to revoke lease: lease_id=auth/cert/login ***** 

**Expected behavior**
token revoked, without any error 
For example 
2022-12-11T20:22:27.714Z [INFO]  expiration: revoked lease: lease_id=auth/cert/login/h2e1b24fb284004483ea5110449916305107dd93387a5edeec7b5e141d10802cf

**Environment:**
* Vault Server Version (retrieve with `vault status`):
   Key             Value
---             -----
Seal Type       shamir
Initialized     true
Sealed          false
Total Shares    1
Threshold       1
Version         1.12.2
Storage Type    postgresql
* Vault CLI Version (retrieve with `vault version`):
Vault v1.12.2
* Server Operating System/Architecture:
docker ubuntu image 

/etc/os-release 
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

Vault server configuration file(s):

```hcl
listener "tcp" {
  address       = "0.0.0.0:8200"
  tls_cert_file = "/tmp/vault.pem"
  tls_key_file  = "/tmp/vault-key.pem"
  tls_disable   = 0
}

api_addr     = "https://0.0.0.0:8200"
cluster_addr = "https://0.0.0.0:8201"

storage "postgresql" {
  connection_url = "*******"
  ha_enabled     = false
}
```

--- a/changelog/19416.txt
+++ b/changelog/19416.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+auth/token: Fix cubbyhole and revocation for legacy service tokens
+```

--- a/vault/token_store.go
+++ b/vault/token_store.go
@@ -1105,7 +1105,7 @@ func (ts *TokenStore) create(ctx context.Context, entry *logical.TokenEntry) err
 			entry.ID = fmt.Sprintf("%s.%s", entry.ID, tokenNS.ID)
 		}
 
-		if tokenNS.ID != namespace.RootNamespaceID || strings.HasPrefix(entry.ID, consts.ServiceTokenPrefix) {
+		if tokenNS.ID != namespace.RootNamespaceID || strings.HasPrefix(entry.ID, consts.ServiceTokenPrefix) || strings.HasPrefix(entry.ID, consts.LegacyServiceTokenPrefix) {
 			if entry.CubbyholeID == "" {
 				cubbyholeID, err := base62.Random(TokenLength)
 				if err != nil {

--- a/vault/token_store_test.go
+++ b/vault/token_store_test.go
@@ -50,11 +50,15 @@ func TestTokenStore_CreateOrphanResponse(t *testing.T) {
 	}
 }
 
+// TestTokenStore_CubbyholeDeletion tests that a token's cubbyhole
+// can be used and that the cubbyhole is removed after the token is revoked.
 func TestTokenStore_CubbyholeDeletion(t *testing.T) {
 	c, _, root := TestCoreUnsealed(t)
 	testTokenStore_CubbyholeDeletion(t, c, root)
 }
 
+// TestTokenStore_CubbyholeDeletionSSCTokensDisabled tests that a legacy token's
+// cubbyhole can be used, and that the cubbyhole is removed after the token is revoked.
 func TestTokenStore_CubbyholeDeletionSSCTokensDisabled(t *testing.T) {
 	c, _, root := TestCoreUnsealed(t)
 	c.disableSSCTokens = true

--- a/vault/token_store_test.go
+++ b/vault/token_store_test.go
@@ -52,6 +52,16 @@ func TestTokenStore_CreateOrphanResponse(t *testing.T) {
 
 func TestTokenStore_CubbyholeDeletion(t *testing.T) {
 	c, _, root := TestCoreUnsealed(t)
+	testTokenStore_CubbyholeDeletion(t, c, root)
+}
+
+func TestTokenStore_CubbyholeDeletionSSCTokensDisabled(t *testing.T) {
+	c, _, root := TestCoreUnsealed(t)
+	c.disableSSCTokens = true
+	testTokenStore_CubbyholeDeletion(t, c, root)
+}
+
+func testTokenStore_CubbyholeDeletion(t *testing.T, c *Core, root string) {
 	ts := c.tokenStore
 
 	for i := 0; i < 10; i++ {

